pom.xml

```diff
@@ -277,6 +301,8 @@
       <id>release</id>
       <properties>
         <karaf.itest.skip>true</karaf.itest.skip>
+        <gpg.executable>gpg</gpg.executable>
```

Comment:
Any reason to move this from maven `settings.xml` to `pom.xml`?

Reply:
This is used to sign artifacts, right now build is breaking due to no gpg keys configured, I was wondering if this could be moved to some other profile rather than being executed in main build

Reply:
@hrishin Didn't work without this for me.

@rohanKanojia 
> I was wondering if this could be moved to some other profile rather than being executed in main build

why?

Reply:
because I don't think we want to sign artifacts everytime we build project

Reply:
we can segregate using the profile that is one option.however its better to keep it into `.settings.xml`
In that way if credentials got compromised we can update maven settings and it doesn't expose the credentials details.

Reply:
>  Didn't work without this for me

Could you please share the build logs whenever possible?

Reply:
```
[kishansagathiya@localhost kubernetes-client]$ mvn clean -B && mvn -V -B -e -U install org.sonatype.plugins:nexus-staging-maven-plugin:1.6.7:deploy -P release -DnexusUrl=https://oss.sonatype.org -DserverId=oss-sonatype-staging
[INFO] Scanning for projects...
[INFO] Inspecting build with total of 12 modules...
[INFO] Installing Nexus Staging features:
[INFO]   ... total of 12 executions of maven-deploy-plugin replaced with nexus-staging-maven-plugin
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Build Order:
[INFO] 
[INFO] Fabric8 :: Kubernetes :: Project
[INFO] Fabric8 :: Kubernetes :: Java Client
[INFO] Fabric8 :: Kubernetes :: Server Mock
[INFO] Fabric8 :: Openshift :: Java Client
[INFO] Fabric8 :: Openshift :: Server Mock
[INFO] Fabric8 :: Kubernetes :: Client Examples
[INFO] Fabric8 :: Kubernetes :: Platforms
[INFO] Fabric8 :: Kubernetes :: Platforms :: Karaf
[INFO] Fabric8 :: Kubernetes :: Platforms :: Karaf :: Features
[INFO] Fabric8 :: Kubernetes :: Platforms :: Karaf :: Tests
[INFO] Fabric8 :: Kubernetes :: Tests
[INFO] Fabric8 :: Kubernetes and Openshift :: UberJar
[INFO] 
[INFO] ------------------------------------------------------------------------
[INFO] Building Fabric8 :: Kubernetes :: Project 4.0-SNAPSHOT
[INFO] ------------------------------------------------------------------------
[INFO] 
[INFO] --- maven-clean-plugin:2.5:clean (default-clean) @ kubernetes-client-project ---
[INFO] Deleting /home/kishansagathiya/redhat/kubernetes-client/target
[INFO] 
[INFO] ------------------------------------------------------------------------
[INFO] Building Fabric8 :: Kubernetes :: Java Client 4.0-SNAPSHOT
[INFO] ------------------------------------------------------------------------
[INFO] 
[INFO] --- maven-clean-plugin:2.5:clean (default-clean) @ kubernetes-client ---
[INFO] Deleting /home/kishansagathiya/redhat/kubernetes-client/kubernetes-client/target
[INFO] 
[INFO] ------------------------------------------------------------------------
[INFO] Building Fabric8 :: Kubernetes :: Server Mock 4.0-SNAPSHOT
[INFO] ------------------------------------------------------------------------
[INFO] 
[INFO] --- maven-clean-plugin:2.5:clean (default-clean) @ kubernetes-server-mock ---
[INFO] Deleting /home/kishansagathiya/redhat/kubernetes-client/kubernetes-server-mock/target
[INFO] 
[INFO] ------------------------------------------------------------------------
[INFO] Building Fabric8 :: Openshift :: Java Client 4.0-SNAPSHOT
[INFO] ------------------------------------------------------------------------
[INFO] 
[INFO] --- maven-clean-plugin:2.5:clean (default-clean) @ openshift-client ---
[INFO] Deleting /home/kishansagathiya/redhat/kubernetes-client/openshift-client/target
[INFO] 
[INFO] ------------------------------------------------------------------------
[INFO] Building Fabric8 :: Openshift :: Server Mock 4.0-SNAPSHOT
[INFO] ------------------------------------------------------------------------
[INFO] 
[INFO] --- maven-clean-plugin:2.5:clean (default-clean) @ openshift-server-mock ---
[INFO] Deleting /home/kishansagathiya/redhat/kubernetes-client/openshift-server-mock/target
[INFO] 
[INFO] ------------------------------------------------------------------------
[INFO] Building Fabric8 :: Kubernetes :: Client Examples 4.0-SNAPSHOT
[INFO] ------------------------------------------------------------------------
[INFO] 
[INFO] --- maven-clean-plugin:2.5:clean (default-clean) @ kubernetes-examples ---
[INFO] Deleting /home/kishansagathiya/redhat/kubernetes-client/kubernetes-examples/target
[INFO] 
[INFO] ------------------------------------------------------------------------
[INFO] Building Fabric8 :: Kubernetes :: Platforms 4.0-SNAPSHOT
[INFO] ------------------------------------------------------------------------
[INFO] 
[INFO] --- maven-clean-plugin:2.5:clean (default-clean) @ platforms ---
[INFO] Deleting /home/kishansagathiya/redhat/kubernetes-client/platforms/target
[INFO] 
[INFO] ------------------------------------------------------------------------
[INFO] Building Fabric8 :: Kubernetes :: Platforms :: Karaf 4.0-SNAPSHOT
[INFO] ------------------------------------------------------------------------
[INFO] 
[INFO] --- maven-clean-plugin:2.5:clean (default-clean) @ karaf ---
[INFO] Deleting /home/kishansagathiya/redhat/kubernetes-client/platforms/karaf/target
[INFO] 
[INFO] ------------------------------------------------------------------------
[INFO] Building Fabric8 :: Kubernetes :: Platforms :: Karaf :: Features 4.0-SNAPSHOT
[INFO] ------------------------------------------------------------------------
[INFO] 
[INFO] --- maven-clean-plugin:2.5:clean (default-clean) @ kubernetes-karaf ---
[INFO] Deleting /home/kishansagathiya/redhat/kubernetes-client/platforms/karaf/features/target
[INFO] 
[INFO] ------------------------------------------------------------------------
[INFO] Building Fabric8 :: Kubernetes :: Platforms :: Karaf :: Tests 4.0-SNAPSHOT
[INFO] ------------------------------------------------------------------------
[INFO] 
[INFO] --- maven-clean-plugin:2.5:clean (default-clean) @ kubernetes-karaf-itests ---
[INFO] Deleting /home/kishansagathiya/redhat/kubernetes-client/platforms/karaf/itests/target
[INFO] 
[INFO] ------------------------------------------------------------------------
[INFO] Building Fabric8 :: Kubernetes :: Tests 4.0-SNAPSHOT
[INFO] ------------------------------------------------------------------------
[INFO] 
[INFO] --- maven-clean-plugin:2.5:clean (default-clean) @ kubernetes-test ---
[INFO] Deleting /home/kishansagathiya/redhat/kubernetes-client/kubernetes-tests/target
[INFO] 
[INFO] ------------------------------------------------------------------------
[INFO] Building Fabric8 :: Kubernetes and Openshift :: UberJar 4.0-SNAPSHOT
[INFO] ------------------------------------------------------------------------
[INFO] 
[INFO] --- maven-clean-plugin:2.5:clean (default-clean) @ kubernetes-openshift-uberjar ---
[INFO] Deleting /home/kishansagathiya/redhat/kubernetes-client/uberjar/target
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary:
[INFO] 
[INFO] Fabric8 :: Kubernetes :: Project ................... SUCCESS [  0.149 s]
[INFO] Fabric8 :: Kubernetes :: Java Client ............... SUCCESS [  0.177 s]
[INFO] Fabric8 :: Kubernetes :: Server Mock ............... SUCCESS [  0.061 s]
[INFO] Fabric8 :: Openshift :: Java Client ................ SUCCESS [  0.092 s]
[INFO] Fabric8 :: Openshift :: Server Mock ................ SUCCESS [  0.013 s]
[INFO] Fabric8 :: Kubernetes :: Client Examples ........... SUCCESS [  0.039 s]
[INFO] Fabric8 :: Kubernetes :: Platforms ................. SUCCESS [  0.025 s]
[INFO] Fabric8 :: Kubernetes :: Platforms :: Karaf ........ SUCCESS [  0.003 s]
[INFO] Fabric8 :: Kubernetes :: Platforms :: Karaf :: Features SUCCESS [  0.004 s]
[INFO] Fabric8 :: Kubernetes :: Platforms :: Karaf :: Tests SUCCESS [  0.031 s]
[INFO] Fabric8 :: Kubernetes :: Tests ..................... SUCCESS [  0.009 s]
[INFO] Fabric8 :: Kubernetes and Openshift :: UberJar ..... SUCCESS [  0.042 s]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 2.075 s
[INFO] Finished at: 2018-09-17T12:15:39+05:30
[INFO] Final Memory: 16M/298M
[INFO] ------------------------------------------------------------------------
Apache Maven 3.5.0 (Red Hat 3.5.0-6)
Maven home: /usr/share/maven
Java version: 1.8.0_161, vendor: Oracle Corporation
Java home: /usr/local/java/jdk1.8.0_161/jre
Default locale: en_US, platform encoding: UTF-8
OS name: "linux", version: "4.13.9-300.fc27.x86_64", arch: "amd64", family: "unix"
[INFO] Error stacktraces are turned on.
[INFO] Scanning for projects...
[INFO] Inspecting build with total of 12 modules...
[INFO] Installing Nexus Staging features:
[INFO]   ... total of 12 executions of maven-deploy-plugin replaced with nexus-staging-maven-plugin
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Build Order:
[INFO] 
[INFO] Fabric8 :: Kubernetes :: Project
[INFO] Fabric8 :: Kubernetes :: Java Client
[INFO] Fabric8 :: Kubernetes :: Server Mock
[INFO] Fabric8 :: Openshift :: Java Client
[INFO] Fabric8 :: Openshift :: Server Mock
[INFO] Fabric8 :: Kubernetes :: Client Examples
[INFO] Fabric8 :: Kubernetes :: Platforms
[INFO] Fabric8 :: Kubernetes :: Platforms :: Karaf
[INFO] Fabric8 :: Kubernetes :: Platforms :: Karaf :: Features
[INFO] Fabric8 :: Kubernetes :: Platforms :: Karaf :: Tests
[INFO] Fabric8 :: Kubernetes :: Tests
[INFO] Fabric8 :: Kubernetes and Openshift :: UberJar
[INFO] 
[INFO] ------------------------------------------------------------------------
[INFO] Building Fabric8 :: Kubernetes :: Project 4.0-SNAPSHOT
[INFO] ------------------------------------------------------------------------
[INFO] 
[INFO] --- sundr-maven-plugin:0.3.12:generate-bom (default) @ kubernetes-client-project ---
[INFO] Downloading: https://repo.maven.apache.org/maven2/org/slf4j/slf4j-simple/1.7.13/slf4j-simple-1.7.13
[INFO] Downloading: https://repo.maven.apache.org/maven2/org/osgi/org.osgi.core/6.0.0/org.osgi.core-6.0.0
[INFO] Downloading: https://repo.maven.apache.org/maven2/io/fabric8/mockwebserver/0.1.0/mockwebserver-0.1.0
[INFO] Downloading: https://repo.maven.apache.org/maven2/org/apache/karaf/itests/itests/4.0.8/itests-4.0.8
[INFO] Downloading: https://repo.maven.apache.org/maven2/junit/junit/4.12/junit-4.12
[INFO] Downloading: https://repo.maven.apache.org/maven2/org/apache/karaf/apache-karaf-minimal/4.0.8/apache-karaf-minimal-4.0.8
[INFO] Downloading: https://repo.maven.apache.org/maven2/org/ops4j/pax/exam/pax-exam-container-karaf/4.9.2/pax-exam-container-karaf-4.9.2
[INFO] Downloading: https://repo.maven.apache.org/maven2/org/ops4j/pax/exam/pax-exam-junit4/4.9.2/pax-exam-junit4-4.9.2
[INFO] Downloading: https://repo.maven.apache.org/maven2/org/apache/geronimo/specs/geronimo-atinject_1.0_spec/1.0/geronimo-atinject_1.0_spec-1.0
[INFO] Downloading: https://repo.maven.apache.org/maven2/org/assertj/assertj-core/3.8.0/assertj-core-3.8.0
[INFO] Downloading: https://repo.maven.apache.org/maven2/io/fabric8/kubernetes-model/3.0.2/kubernetes-model-3.0.2
[INFO] Downloading: https://repo.maven.apache.org/maven2/com/fasterxml/jackson/core/jackson-core/2.7.7/jackson-core-2.7.7
[INFO] Downloading: https://repo.maven.apache.org/maven2/com/squareup/okhttp3/okhttp/3.9.1/okhttp-3.9.1
[INFO] Downloading: https://repo.maven.apache.org/maven2/javax/validation/validation-api/1.1.0.Final/validation-api-1.1.0.Final
[INFO] Downloading: https://repo.maven.apache.org/maven2/com/fasterxml/jackson/core/jackson-databind/2.7.7/jackson-databind-2.7.7
[INFO] Downloading: https://repo.maven.apache.org/maven2/com/squareup/okhttp3/logging-interceptor/3.9.1/logging-interceptor-3.9.1
[INFO] Downloading: https://repo.maven.apache.org/maven2/org/slf4j/slf4j-api/1.7.13/slf4j-api-1.7.13
[INFO] Downloading: https://repo.maven.apache.org/maven2/org/slf4j/jul-to-slf4j/1.7.13/jul-to-slf4j-1.7.13
[INFO] Downloading: https://repo.maven.apache.org/maven2/com/fasterxml/jackson/dataformat/jackson-dataformat-yaml/2.7.7/jackson-dataformat-yaml-2.7.7
[INFO] Downloading: https://repo.maven.apache.org/maven2/io/fabric8/zjsonpatch/0.3.0/zjsonpatch-0.3.0
[INFO] Downloading: https://repo.maven.apache.org/maven2/com/github/mifmif/generex/1.0.1/generex-1.0.1
[INFO] Downloading: https://repo.maven.apache.org/maven2/io/sundr/builder-annotations/0.3.12/builder-annotations-0.3.12
[INFO] Downloading: https://repo.maven.apache.org/maven2/org/apache/felix/org.apache.felix.scr.annotations/1.9.12/org.apache.felix.scr.annotations-1.9.12
[INFO] Downloading: https://repo.maven.apache.org/maven2/org/apache/logging/log4j/log4j-slf4j-impl/2.10.0/log4j-slf4j-impl-2.10.0
[INFO] Downloading: https://repo.maven.apache.org/maven2/org/apache/logging/log4j/log4j-core/2.10.0/log4j-core-2.10.0
[INFO] Generating BOM: kubernetes-client-bom
[WARNING] The BOM kubernetes-client-bom-with-deps contains multiple versions of the following dependencies:
 - com.fasterxml.jackson.core:jackson-databind:jar:null versions [2.7.0, 2.7.7]

[INFO] Generating BOM: kubernetes-client-bom-with-deps
[INFO] 
[INFO] ------------------------------------------------------------------------
[INFO] Building Fabric8 :: Kubernetes :: Bom 4.0-SNAPSHOT
[INFO] ------------------------------------------------------------------------
[INFO] 
[INFO] --- sundr-maven-plugin:0.3.12:generate-bom (default) @ kubernetes-client-bom ---
[INFO] 
[INFO] --- build-helper-maven-plugin:1.10:parse-version (parse-version) @ kubernetes-client-bom ---
[INFO] 
[INFO] --- maven-enforcer-plugin:1.3.1:enforce (enforce-maven) @ kubernetes-client-bom ---
[INFO] 
[INFO] --- maven-enforcer-plugin:1.3.1:enforce (enforce-no-snapshots) @ kubernetes-client-bom ---
[INFO] 
[INFO] --- jacoco-maven-plugin:0.7.5.201505241946:prepare-agent (default-prepare-agent) @ kubernetes-client-bom ---
[INFO] argLine set to -javaagent:/home/kishansagathiya/.m2/repository/org/jacoco/org.jacoco.agent/0.7.5.201505241946/org.jacoco.agent-0.7.5.201505241946-runtime.jar=destfile=/home/kishansagathiya/redhat/kubernetes-client/target/classes/kubernetes-client-bom/jacoco.exec
[INFO] 
[INFO] --- maven-javadoc-plugin:2.10.3:jar (attach-javadocs) @ kubernetes-client-bom ---
[INFO] Not executing Javadoc as the project is not a Java classpath-capable package
[INFO] 
[INFO] >>> maven-source-plugin:2.4:jar (attach-sources) > generate-sources @ kubernetes-client-bom >>>
[INFO] 
[INFO] --- sundr-maven-plugin:0.3.12:generate-bom (default) @ kubernetes-client-bom ---
[INFO] 
[INFO] --- build-helper-maven-plugin:1.10:parse-version (parse-version) @ kubernetes-client-bom ---
[INFO] 
[INFO] --- maven-enforcer-plugin:1.3.1:enforce (enforce-maven) @ kubernetes-client-bom ---
[INFO] 
[INFO] --- maven-enforcer-plugin:1.3.1:enforce (enforce-no-snapshots) @ kubernetes-client-bom ---
[INFO] 
[INFO] --- jacoco-maven-plugin:0.7.5.201505241946:prepare-agent (default-prepare-agent) @ kubernetes-client-bom ---
[INFO] argLine set to -javaagent:/home/kishansagathiya/.m2/repository/org/jacoco/org.jacoco.agent/0.7.5.201505241946/org.jacoco.agent-0.7.5.201505241946-runtime.jar=destfile=/home/kishansagathiya/redhat/kubernetes-client/target/classes/kubernetes-client-bom/jacoco.exec
[INFO] 
[INFO] <<< maven-source-plugin:2.4:jar (attach-sources) < generate-sources @ kubernetes-client-bom <<<
[INFO] 
[INFO] 
[INFO] --- maven-source-plugin:2.4:jar (attach-sources) @ kubernetes-client-bom ---
[INFO] 
[INFO] --- jacoco-maven-plugin:0.7.5.201505241946:report (default-report) @ kubernetes-client-bom ---
[INFO] Skipping JaCoCo execution due to missing execution data file:/home/kishansagathiya/redhat/kubernetes-client/target/classes/kubernetes-client-bom/jacoco.exec
[INFO] 
[INFO] --- maven-gpg-plugin:1.5:sign (sign-artifacts) @ kubernetes-client-bom ---
gpg: gpg-agent is not available in this session
gpg: Sorry, no terminal at all requested - can't get input
[INFO] 
[INFO] ------------------------------------------------------------------------
[INFO] Building Fabric8 :: Kubernetes :: Bom with Dependencies 4.0-SNAPSHOT
[INFO] ------------------------------------------------------------------------
[INFO] 
[INFO] --- sundr-maven-plugin:0.3.12:generate-bom (default) @ kubernetes-client-bom-with-deps ---
[INFO] 
[INFO] --- build-helper-maven-plugin:1.10:parse-version (parse-version) @ kubernetes-client-bom-with-deps ---
[INFO] 
[INFO] --- maven-enforcer-plugin:1.3.1:enforce (enforce-maven) @ kubernetes-client-bom-with-deps ---
[INFO] 
[INFO] --- maven-enforcer-plugin:1.3.1:enforce (enforce-no-snapshots) @ kubernetes-client-bom-with-deps ---
[INFO] 
[INFO] --- jacoco-maven-plugin:0.7.5.201505241946:prepare-agent (default-prepare-agent) @ kubernetes-client-bom-with-deps ---
[INFO] argLine set to -javaagent:/home/kishansagathiya/.m2/repository/org/jacoco/org.jacoco.agent/0.7.5.201505241946/org.jacoco.agent-0.7.5.201505241946-runtime.jar=destfile=/home/kishansagathiya/redhat/kubernetes-client/target/classes/kubernetes-client-bom-with-deps/jacoco.exec
[INFO] 
[INFO] --- maven-javadoc-plugin:2.10.3:jar (attach-javadocs) @ kubernetes-client-bom-with-deps ---
[INFO] Not executing Javadoc as the project is not a Java classpath-capable package
[INFO] 
[INFO] >>> maven-source-plugin:2.4:jar (attach-sources) > generate-sources @ kubernetes-client-bom-with-deps >>>
[INFO] 
[INFO] --- sundr-maven-plugin:0.3.12:generate-bom (default) @ kubernetes-client-bom-with-deps ---
[INFO] 
[INFO] --- build-helper-maven-plugin:1.10:parse-version (parse-version) @ kubernetes-client-bom-with-deps ---
[INFO] 
[INFO] --- maven-enforcer-plugin:1.3.1:enforce (enforce-maven) @ kubernetes-client-bom-with-deps ---
[INFO] 
[INFO] --- maven-enforcer-plugin:1.3.1:enforce (enforce-no-snapshots) @ kubernetes-client-bom-with-deps ---
[INFO] 
[INFO] --- jacoco-maven-plugin:0.7.5.201505241946:prepare-agent (default-prepare-agent) @ kubernetes-client-bom-with-deps ---
[INFO] argLine set to -javaagent:/home/kishansagathiya/.m2/repository/org/jacoco/org.jacoco.agent/0.7.5.201505241946/org.jacoco.agent-0.7.5.201505241946-runtime.jar=destfile=/home/kishansagathiya/redhat/kubernetes-client/target/classes/kubernetes-client-bom-with-deps/jacoco.exec
[INFO] 
[INFO] <<< maven-source-plugin:2.4:jar (attach-sources) < generate-sources @ kubernetes-client-bom-with-deps <<<
[INFO] 
[INFO] 
[INFO] --- maven-source-plugin:2.4:jar (attach-sources) @ kubernetes-client-bom-with-deps ---
[INFO] 
[INFO] --- jacoco-maven-plugin:0.7.5.201505241946:report (default-report) @ kubernetes-client-bom-with-deps ---
[INFO] Skipping JaCoCo execution due to missing execution data file:/home/kishansagathiya/redhat/kubernetes-client/target/classes/kubernetes-client-bom-with-deps/jacoco.exec
[INFO] 
[INFO] --- maven-gpg-plugin:1.5:sign (sign-artifacts) @ kubernetes-client-bom-with-deps ---
gpg: gpg-agent is not available in this session
gpg: Sorry, no terminal at all requested - can't get input
[INFO] 
[INFO] --- build-helper-maven-plugin:1.10:parse-version (parse-version) @ kubernetes-client-project ---
[INFO] 
[INFO] --- maven-enforcer-plugin:1.3.1:enforce (enforce-maven) @ kubernetes-client-project ---
[INFO] 
[INFO] --- maven-enforcer-plugin:1.3.1:enforce (enforce-no-snapshots) @ kubernetes-client-project ---
[INFO] 
[INFO] --- jacoco-maven-plugin:0.7.5.201505241946:prepare-agent (default-prepare-agent) @ kubernetes-client-project ---
[INFO] argLine set to -javaagent:/home/kishansagathiya/.m2/repository/org/jacoco/org.jacoco.agent/0.7.5.201505241946/org.jacoco.agent-0.7.5.201505241946-runtime.jar=destfile=/home/kishansagathiya/redhat/kubernetes-client/target/jacoco.exec
[INFO] 
[INFO] --- maven-javadoc-plugin:2.10.3:jar (attach-javadocs) @ kubernetes-client-project ---
[INFO] Not executing Javadoc as the project is not a Java classpath-capable package
[INFO] 
[INFO] >>> maven-source-plugin:2.4:jar (attach-sources) > generate-sources @ kubernetes-client-project >>>
[INFO] 
[INFO] --- sundr-maven-plugin:0.3.12:generate-bom (default) @ kubernetes-client-project ---
[INFO] 
[INFO] --- build-helper-maven-plugin:1.10:parse-version (parse-version) @ kubernetes-client-project ---
[INFO] 
[INFO] --- maven-enforcer-plugin:1.3.1:enforce (enforce-maven) @ kubernetes-client-project ---
[INFO] 
[INFO] --- maven-enforcer-plugin:1.3.1:enforce (enforce-no-snapshots) @ kubernetes-client-project ---
[INFO] 
[INFO] --- jacoco-maven-plugin:0.7.5.201505241946:prepare-agent (default-prepare-agent) @ kubernetes-client-project ---
[INFO] argLine set to -javaagent:/home/kishansagathiya/.m2/repository/org/jacoco/org.jacoco.agent/0.7.5.201505241946/org.jacoco.agent-0.7.5.201505241946-runtime.jar=destfile=/home/kishansagathiya/redhat/kubernetes-client/target/jacoco.exec
[INFO] 
[INFO] <<< maven-source-plugin:2.4:jar (attach-sources) < generate-sources @ kubernetes-client-project <<<
[INFO] 
[INFO] 
[INFO] --- maven-source-plugin:2.4:jar (attach-sources) @ kubernetes-client-project ---
[INFO] 
[INFO] --- jacoco-maven-plugin:0.7.5.201505241946:report (default-report) @ kubernetes-client-project ---
[INFO] Skipping JaCoCo execution due to missing execution data file:/home/kishansagathiya/redhat/kubernetes-client/target/jacoco.exec
[INFO] 
[INFO] --- maven-gpg-plugin:1.5:sign (sign-artifacts) @ kubernetes-client-project ---
gpg: gpg-agent is not available in this session
gpg: Sorry, no terminal at all requested - can't get input
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary:
[INFO] 
[INFO] Fabric8 :: Kubernetes :: Project ................... FAILURE [ 14.134 s]
[INFO] Fabric8 :: Kubernetes :: Java Client ............... SKIPPED
[INFO] Fabric8 :: Kubernetes :: Server Mock ............... SKIPPED
[INFO] Fabric8 :: Openshift :: Java Client ................ SKIPPED
[INFO] Fabric8 :: Openshift :: Server Mock ................ SKIPPED
[INFO] Fabric8 :: Kubernetes :: Client Examples ........... SKIPPED
[INFO] Fabric8 :: Kubernetes :: Platforms ................. SKIPPED
[INFO] Fabric8 :: Kubernetes :: Platforms :: Karaf ........ SKIPPED
[INFO] Fabric8 :: Kubernetes :: Platforms :: Karaf :: Features SKIPPED
[INFO] Fabric8 :: Kubernetes :: Platforms :: Karaf :: Tests SKIPPED
[INFO] Fabric8 :: Kubernetes :: Tests ..................... SKIPPED
[INFO] Fabric8 :: Kubernetes and Openshift :: UberJar ..... SKIPPED
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 15.613 s
[INFO] Finished at: 2018-09-17T12:15:58+05:30
[INFO] Final Memory: 37M/539M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-gpg-plugin:1.5:sign (sign-artifacts) on project kubernetes-client-bom: Exit code: 2 -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal org.apache.maven.plugins:maven-gpg-plugin:1.5:sign (sign-artifacts) on project kubernetes-client-bom: Exit code: 2
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:213)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:154)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:146)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:117)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:81)
	at io.sundr.maven.GenerateBomMojo.build(GenerateBomMojo.java:374)
	at io.sundr.maven.GenerateBomMojo.execute(GenerateBomMojo.java:148)
	at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:134)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:208)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:154)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:146)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:117)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:81)
	at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build(SingleThreadedBuilder.java:51)
	at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:128)
	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:309)
	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:194)
	at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:107)
	at org.apache.maven.cli.MavenCli.execute(MavenCli.java:993)
	at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:345)
	at org.apache.maven.cli.MavenCli.main(MavenCli.java:191)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:289)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:229)
	at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:415)
	at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:356)
Caused by: org.apache.maven.plugin.MojoExecutionException: Exit code: 2
	at org.apache.maven.plugin.gpg.GpgSigner.generateSignatureForFile(GpgSigner.java:168)
	at org.apache.maven.plugin.gpg.AbstractGpgSigner.generateSignatureForArtifact(AbstractGpgSigner.java:205)
	at org.apache.maven.plugin.gpg.GpgSignAttachedMojo.execute(GpgSignAttachedMojo.java:176)
	at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:134)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:208)
	... 28 more
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-gpg-plugin:1.5:sign (sign-artifacts) on project kubernetes-client-bom-with-deps: Exit code: 2 -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal org.apache.maven.plugins:maven-gpg-plugin:1.5:sign (sign-artifacts) on project kubernetes-client-bom-with-deps: Exit code: 2
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:213)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:154)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:146)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:117)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:81)
	at io.sundr.maven.GenerateBomMojo.build(GenerateBomMojo.java:374)
	at io.sundr.maven.GenerateBomMojo.execute(GenerateBomMojo.java:148)
	at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:134)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:208)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:154)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:146)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:117)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:81)
	at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build(SingleThreadedBuilder.java:51)
	at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:128)
	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:309)
	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:194)
	at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:107)
	at org.apache.maven.cli.MavenCli.execute(MavenCli.java:993)
	at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:345)
	at org.apache.maven.cli.MavenCli.main(MavenCli.java:191)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:289)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:229)
	at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:415)
	at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:356)
Caused by: org.apache.maven.plugin.MojoExecutionException: Exit code: 2
	at org.apache.maven.plugin.gpg.GpgSigner.generateSignatureForFile(GpgSigner.java:168)
	at org.apache.maven.plugin.gpg.AbstractGpgSigner.generateSignatureForArtifact(AbstractGpgSigner.java:205)
	at org.apache.maven.plugin.gpg.GpgSignAttachedMojo.execute(GpgSignAttachedMojo.java:176)
	at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:134)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:208)
	... 28 more
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-gpg-plugin:1.5:sign (sign-artifacts) on project kubernetes-client-project: Exit code: 2 -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal org.apache.maven.plugins:maven-gpg-plugin:1.5:sign (sign-artifacts) on project kubernetes-client-project: Exit code: 2
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:213)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:154)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:146)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:117)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:81)
	at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build(SingleThreadedBuilder.java:51)
	at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:128)
	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:309)
	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:194)
	at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:107)
	at org.apache.maven.cli.MavenCli.execute(MavenCli.java:993)
	at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:345)
	at org.apache.maven.cli.MavenCli.main(MavenCli.java:191)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:289)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:229)
	at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:415)
	at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:356)
Caused by: org.apache.maven.plugin.MojoExecutionException: Exit code: 2
	at org.apache.maven.plugin.gpg.GpgSigner.generateSignatureForFile(GpgSigner.java:168)
	at org.apache.maven.plugin.gpg.AbstractGpgSigner.generateSignatureForArtifact(AbstractGpgSigner.java:205)
	at org.apache.maven.plugin.gpg.GpgSignAttachedMojo.execute(GpgSignAttachedMojo.java:176)
	at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:134)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:208)
	... 20 more
[ERROR] 
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
[ERROR] 
[ERROR] After correcting the problems, you can resume the build with the command
[ERROR]   mvn <goals> -rf :kubernetes-client-bom
[kishansagathiya@localhost kubernetes-client]$ 

```